core/common/src/main/scala/org/typelevel/otel4s/Attributes.scala

```diff
@@ -112,25 +111,44 @@ sealed trait Attributes
 
   /** Removes all attributes with any of the given keys. */
   def removedAll(that: IterableOnce[AttributeKey[_]]): Attributes =
-    attributesFactory.fromSpecific((toMap -- that).values)
+    that.iterator.foldLeft(this)(_ - _)
 
   /** Removes all attributes with any of the given keys. */
   final def --(that: IterableOnce[AttributeKey[_]]): Attributes =
     removedAll(that)
 
-  /** @return the `Map` representation of these `Attributes` */
-  def toMap: Map[AttributeKey[_], Attribute[_]]
+  /** @return
+    *   the `Map` representation of these `Attributes` with untyped `String`
+    *   keys
+    */
+  def asMap: Map[String, Attribute[_]]
+
+  /** @return
+    *   the `Map` representation of these `Attributes` with untyped `String`
+    *   keys
+    */
+  @deprecated("use `asMap` instead", since = "0.5.0")
+  final def toMap: Map[String, Attribute[_]] = asMap
```

Comment:
We have scalafix rules for the upcoming release. I can write a migration for this case as well. 
So we don't need to keep deprecated methods.

Reply:
especially given that the type signature has changed. yeah, I'll remove it

Reply:
> `Attributes` class wasn't even published yet.

I completely forgot! lol